pkg/agent/containerd/containerd.go

```diff
@@ -175,11 +176,24 @@ func preloadImages(ctx context.Context, cfg *config.Node) error {
 
 	// Ensure that nothing else can modify the image store while we're importing,
 	// and that our images are imported into the k8s.io namespace
-	ctx, done, err := client.WithLease(namespaces.WithNamespace(ctx, "k8s.io"))
+	ctxWithNs := namespaces.WithNamespace(ctx, "k8s.io")
+	// At startup all images in the store with a lease are cleared
+	ls := client.LeasesService()
+	existingLeases, err := ls.List(ctxWithNs)
+	if err != nil {
+		return err
+	}
+
+	for _, lease := range existingLeases {
+		logrus.Debugf("Deleting existing lease: %v", lease)
+		ls.Delete(ctxWithNs, lease)
```

Comment:
Don't delete all leases; we're not guaranteed to be the only thing holding them. Set a lease using our name as the ID, and only delete our leases on startup.
```suggestion
		if lease.ID == version.Program {
			logrus.Debugf("Deleting existing lease: %v", lease)
			ls.Delete(ctx, lease)
		}
```

Reply:
still needs to be addressed